firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexBackfiller.java

```diff
@@ -157,25 +157,24 @@ private IndexOffset getExistingOffset(Collection<FieldIndex> fieldIndexes) {
         lowestOffset = fieldIndex.getIndexState().getOffset();
       }
     }
-    return lowestOffset == null ? IndexOffset.NONE : lowestOffset;
+    lowestOffset = lowestOffset == null ? IndexOffset.NONE : lowestOffset;
+
+    // Add earliest batch id to offset
+    int earliestBatchId = getEarliestBatchId(fieldIndexes);
```

Comment:
Can you inline this call? You are iterating over the list of field indexes twice.